casepro/statistics/views.py

```diff
@@ -4,37 +4,58 @@
 from dash.orgs.views import OrgPermsMixin
 from django.http import JsonResponse
 from django.utils import timezone
-from django.views.generic import View
+from smartmin.views import SmartTemplateView
 
 from casepro.cases.models import Partner
 from casepro.utils import month_range
 
 from .models import DailyCount
 
 
-class PartnerRepliesPerMonthChart(OrgPermsMixin, View):
-    """
-    Chart of replies per month for a specific partner org
-    """
-    permission = 'cases.partner_read'
+class PerMonthChart(OrgPermsMixin, SmartTemplateView):
+    num_months = 12
 
     def get(self, request, *args, **kwargs):
-        partner = Partner.objects.get(org=self.request.org, pk=kwargs['partner_id'])
         now = timezone.now()
 
-        since = month_range(-5)[0]  # last six months ago including this month
-        totals = DailyCount.get_by_partner([partner], DailyCount.TYPE_REPLIES, since).month_totals()
+        since = month_range(-(self.num_months - 1))[0]  # last X months including this month
+        totals = self.get_month_totals(request, since)
         totals_by_month = {t[0]: t[1] for t in totals}
 
         # generate category labels and series over last six months
         categories = []
         series = []
         this_month = now.date().month
-        for m in reversed(range(0, -6, -1)):
+        for m in reversed(range(0, -self.num_months, -1)):
             month = this_month + m
             if month < 1:
                 month += 12
             categories.append(month_name[month])
             series.append(totals_by_month.get(month, 0))
 
         return JsonResponse({'categories': categories, 'series': series})
+
+    def get_month_totals(self, request, since):
+        """
+        Chart classes override this to provide a list of month/value tuples
+        """
+
+
+class RepliesPerMonthChart(PerMonthChart):
+    """
+    Chart of replies per month for either the current org, a given partner, or a given user
+    """
+    permission = 'orgs.org_charts'
```

Comment:
I'm assuming that aggregate data like number of replies can be seen by any user in the org regardless of whether it's for another partner or user. Flagging this in case that's not an assumption that works for other deployments.